### DISCUSSION TITLE
improvement suggestion

### DISCUSSION THREAD

**Discussion by edo-jan** [itcon-pty-au/typingmind-cloud-backup#11] [Ideas]
First of all, thanks a lot for your app. It works great and sync everything to my S3 bucket. However, I am not too clear about the recovery process given that the backup is taken every 1 min.  
Let's take a scenario whereas I have accidently deleted / corrupted certain plugins, etc.  This corruption would immediately (within a minute) be propagated to S3 - correct?  I am wondering how do I restore the backup state to e.g. as of 10 min ago?  I guess, something like a time machine would be helpful to "Import from S3" to be used in conjunction with a time and date option to select.  It appears that currently, the moment .json is overwritten the only one to use would be the one from the previous date. Am I wrong?  

I wonder if there is a simpler way of achieving the same?  Also I would much appreciate if you could shed some light on how this works with the built-in cloud backup feature (10mg of space which comes with the premium version). Currently, my cloud storage is almost full and I am not sure what would be a recommended way to use it in conjunction with your plugin so that both are used without a conflict.

**Comment by itcon-pty-au** [comment]
You are correct, the only option you have is to restore using the previous day's backup. This is mainly for restoring data in case of a catastrophic mistake. The time machine is a good idea. I may not be able to implement it the way you envisioned it. But let me think about what can be done.

This plugin does work along with the built in cloud backup feature. Did you face any conflicts when using both together? However, an export from the inbuilt cloud backup cannot be restored using this extension as the backup file formats are different currently.

**Comment by edo-jan** [comment]
> Did you face any conflicts when using both together? 

No, I am trying to figure out a work around to the 10 MB space limitation in TypingMind cloud. Do you have an idea how this could be done? E.g. To archive some old chats (which I don't want to delete) to S3 and save the cloud space to sync with my other devices. But then I am not clear how to switch back and force between the "production" and "archive" modes

**Reply by itcon-pty-au** [reply]
Trying to understand why you want to have both inbuilt backup and this extension to work in parallel. Won't this extension alone work for you?

**Reply by edo-jan** [reply]
> Trying to understand why you want to have both inbuilt backup and this extension to work in parallel. Won't this extension alone work for you?

You are right. Actually, it would.

**Reply by itcon-pty-au** [reply]
@edo-jan - I have implemented a T-15 backup now. It will create an additional rolling backup of the typingmind data every 15 min. This will give you a secondary backup that you can revert to in case your main backup (every 1 min) file gets corrupted. Note that still you get very less time to download this depending on where you are at in the 15 min cycle. Added a UI based download and restore option to make the restoring quicker.

I have also added a Snapshot functionality to create adhoc 'no-touch' backups whenever you want. Let me know if this helps.

![image](https://github.com/user-attachments/assets/5bf04673-4d31-4222-84c0-2d1a516fcc71)

**Comment by edo-jan** [comment]
This is fantastic, thank you very much! There are a only a couple of issues.  1) The first one is the mouse-over the buttons "Import from S3" and "Snapshot" doesn't work consistently. I have to place the mouse cursor and move it around the button for sometime to find an area where it becomes clickable (e.g. on the Snapshot button it is the right side of the button, and with the "Import from S3" it is the left side . Also, it appears that there is a bit of a lag until it becomes clickable at all. 2) The second one is not an issue but a suggestion to hide (make collapsible) the info about the S3 bucket / AWS region / Keys and perhaps moving this section to sit at the bottom of the dialog - i.e. after the Backup section.   Otherwise, it works great.

**Reply by itcon-pty-au** [reply]
Good catch on the buttons. Fixed it.
Suggestion to move the restore part to the top makes sense as well. This is now implemented.

**Comment by edo-jan** [comment]
Thank you! Works great
